darwinpush/client.py

```diff
@@ -143,15 +193,21 @@ def _connect(self):
         # while self.connected:
         #     time.sleep(1)
 
-    def _on_message(self, headers, message):
+    def on_ftp_message(self, message, source="FTP"):
+        self._on_message(None, message, source)
+
+    def _on_message(self, headers, message, source=None):
+
+        if type(message) == bytes:
+            message = message.decode("utf-8")
 
         # Decode the message and parse it as an XML DOM.
-        doc = domutils.StringToDOM(message.decode("utf-8"))
+        doc = domutils.StringToDOM(message)
```

Comment:
I didn't touch that part. Only the `.decode("utf8")` has changed and moved to stomp, after decompression. I get strings from FTP directly, not bytes.

If it works fine with strings directly than it's fine with me.